### PR TITLE
Migrate test_driver.dart and test_utils.dart

### PR DESCRIPTION
**Pull request by Abhishek01039**
## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by zanderso** [comment]
It looks like this PR is causing failures in some of the pre-submit tests.

File: packages/flutter_tools/test/integration.shard/test_driver.dart [hunk 1]
```diff
@@ -702,14 +701,14 @@ class FlutterRunTestDriver extends FlutterTestDriver {
...
-    _process.stdin.writeln(jsonEncoded);
+    _process?.stdin.writeln(jsonEncoded);
     final Map<String, dynamic> response = await responseFuture;
 
     if (response['error'] != null || response['result'] == null) {
       _throwErrorResponse('Unexpected error response');
     }
 
-    return response['result'];
+    return response['result'] as int?;
```
Comment (Abhishek01039) [inline packages/flutter_tools/test/integration.shard/test_driver.dart:711]: Here we are getting this error
```type 'bool' is not a subtype of type 'int?' in type cast```

Comment (zanderso) [reply packages/flutter_tools/test/integration.shard/test_driver.dart:711]: `int?` is not the right type here. First on line 707 there is already a null check on `response['result']`. Second in the `_restart` method on line 623, the result of `_sendRequest` is assigned to a `Map<String, dynamic>?`.

This PR needs some more thought.

**Review by zanderso — CHANGES_REQUESTED** [review]

File: packages/flutter_tools/test/integration.shard/test_driver.dart (hunk 1 above)
Comment (Abhishek01039) [reply packages/flutter_tools/test/integration.shard/test_driver.dart]: I have make it dynamic.

**Comment by zanderso** [comment]
Will review after presubmit checks are passing.

**Comment by jonahwilliams** [comment]
Because of the lack of activity on this PR I'm going to close it to remove from our review queue. Please feel free to send us a new PR or discuss in an issue anytime.

**Pull request closed without merging**